### PR TITLE
Remove invalid pin_compatible config from defaults

### PR DESCRIPTION
**Pull request by zvecr**
## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: data/mappings/defaults.hjson [hunk 1]
```diff
@@ -8,43 +8,37 @@
         "elite_c": {
             "processor": "atmega32u4",
             "bootloader": "atmel-dfu",
-            "pin_compatible": "promicro"
+            "pin_compatible": "elite_c"
         },
```
Comment (sigprof) [inline data/mappings/defaults.hjson:12]: This is technically correct, but would change the behavior for some existing boards that have `"development_board": "elite_c"`:
```
keyboards/argo_works/ishi/80/mk0_avr_extra/info.json:    "development_board": "elite_c",
keyboards/boardsource/lulu/avr/info.json:  "development_board": "elite_c",
keyboards/handwired/onekey/elite_c/info.json:    "development_board": "elite_c",
keyboards/splitkb/aurora/corne/info.json:    "development_board": "elite_c",
keyboards/splitkb/aurora/lily58/info.json:    "development_board": "elite_c",
keyboards/splitkb/aurora/sweep/info.json:    "development_board": "elite_c",
```
- `argo_works/ishi/80/mk0_avr_extra` actually uses the extra Elite-C pins, and this fix would actually allow removing the `"pin_compatible": "elite_c"` override from that board (and another `PIN_COMPATIBLE = elite_c` override in `rules.mk` which was added as a workaround for a bug that's getting fixed in #19511).
- `boardsource/lulu/avr` uses pins `F0` and `F1`, so `"pin_compatible": "elite_c"` would be correct for that board.
- `handwired/onekey/elite_c` is obviously Elite-C.
- `splitkb/aurora/corne`, `splitkb/aurora/lily58` and `splitkb/aurora/sweep` apparently use only the Pro Micro pins (the PCBs don't even have holes for extra Elite-C pins), so these boards would need to have `"pin_compatible": "promicro"` added (and #19511 to make that actually work).

**Review by sigprof — APPROVED** [review]

File: data/mappings/defaults.hjson (hunk 1 above)
Comment (sigprof) [reply data/mappings/defaults.hjson:12]: Also if this fix goes into master without #19511, the mentioned splitkb boards would need to have `PIN_COMPATIBLE = promicro` added to `rules.mk` instead.

Comment (zvecr) [reply data/mappings/defaults.hjson:12]: Ah good point, lets keep it as is for now.

File: data/mappings/defaults.hjson [hunk 2]
```diff
@@ -8,43 +8,37 @@
         "elite_c": {
             "processor": "atmega32u4",
             "bootloader": "atmel-dfu",
-            "pin_compatible": "promicro"
+            "pin_compatible": "elite_c"
```
Comment (zvecr) [inline data/mappings/defaults.hjson:11]:
```suggestion
            "pin_compatible": "promicro"
```

**Comment by sigprof** [comment]
What do you plan to do about Elite-C — update its defaults some time later in develop and fix the fallout there?  Having `"pin_compatible": "promicro"` for Elite-C is wrong in general, but it is not possible to detect automatically that some keyboard has `"development_board": "elite_c"` only because its maintainer wants the default firmware to match the controller that is commonly sold together with the board.  Or should those cases use `"development_board": "promicro", "bootloader": "atmel-dfu"` instead (which would require #19511 to work)?

(Also it's sad to see the loss of information like https://github.com/qmk/qmk_firmware/pull/18880#discussion_r1008407799 only because `info.json` is named as if it actually was JSON.)

The rest of this PR looks correct though.

**Comment by zvecr** [comment]
The long term plan is potentially that the `elite_c` converters can fulfil `promicro` targets too? Some form of inheritance on the imaginary interfaces they provide. Few issues stopping that at the moment that myself/someone will have to work through.

Short term, I guess i'm not a fan of `"development_board": "elite_c"` then being used with `promicro` converters.  However as all it does is set bootloader, which the converters will then override, I cant see it doing too much harm.

**Pull request merged**